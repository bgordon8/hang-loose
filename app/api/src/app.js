import express from 'express'

const app = express()

app.get("/", (req, res) => {
    res.status(200).json({
        message: "hello world!!!! thats whats up!!!"
    })
})

if (process.env.NODE_ENV !== "test") {
    app.listen(3000, () => {
        console.log('listening on port 3000...')
    })
}

export default app