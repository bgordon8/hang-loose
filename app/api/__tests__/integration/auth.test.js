import supertest from 'supertest'
import app from '../../src/app'

const req = supertest(app)

describe('routes : auth', () => {
  describe('POST /auth/register', () => {
    it('registers a new user and returns a token', async () => {
      const res = await req.post('/auth/register').send({
        email: 'test@email.com',
        username: 'testUser',
        password: 'password123',
      })

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('token')
      expect(res.body).toMatchSnapshot({
        token: expect.any(String),
      })
    })
  })

  describe('POST /auth/login', () => {
    it('logs in a user and returns a token', async () => {
      const res = await req.post('/auth/login').send({
        email: 'leo@email.com',
        password: 'password123',
      })

      console.log(res.body)
      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('token')
      expect(res.body).toMatchSnapshot({
        token: expect.any(String),
      })
    })
  })
})
