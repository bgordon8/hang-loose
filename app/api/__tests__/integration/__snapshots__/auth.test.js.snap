// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes : auth POST /auth/login logs in a user and returns a token 1`] = `
Object {
  "status": "success",
  "token": Any<String>,
}
`;

exports[`routes : auth POST /auth/register registers a new user and returns a token 1`] = `
Object {
  "status": "success",
  "token": Any<String>,
}
`;
